PerfettoCds/Tools/PerfettoTraceProcessor.cs

```diff
@@ -0,0 +1,175 @@
+﻿using Google.Protobuf;
+using Perfetto.Protos;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace PerfettoCds
+{
+    /// <summary>
+    /// Responsible for managing trace_processor_shell.exe. The shell executable loads the Perfetto trace and allows for SQL querying
+    /// through HTTP on localhost. HTTP communication happens with protobuf objects/buffers.
+    /// </summary>
+    public class PerfettoTraceProcessor
+    {
+        private Process ShellProcess;
+
+        // Default starting port
+        private int HttpPort = 22222;
+
+        // Highest ports can go
+        private const int PortMax = 65535;
+
+        // HTTP request denied takes about 3 seconds so time out after about 2 minutes
+        private const int MaxRetryLimit = 40; 
+
+        /// <summary>
+        /// Check if another instance of trace_processor_shell is running on this port
+        /// </summary>
+        /// <param name="port"></param>
+        /// <returns></returns>
+        private bool IsPortAvailable(int port)
+        {
+            try
+            {
+                using (var client = new HttpClient())
+                {
+                    var response = client.GetStringAsync($"http://localhost:{HttpPort}/status").Result;
+                    return !(response.Contains("Perfetto"));
+                }
+            }
+            catch (Exception)
+            {
+                // Exception here is the connection refused message, meaning the RPC server has not been initialized
+                // Which means this port is not being used by another trace_processor_shell
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Initializes trace_processor_shell.exe in HTTP/RPC mode with the trace file
+        /// </summary>
+        /// <param name="tracePath"></param>
+        public void OpenTraceProcessor(string tracePath)
+        {
+            using (var client = new HttpClient())
+            {
+                // Make sure another instance of trace_processor_shell isn't already running
+                while(!IsPortAvailable(HttpPort) && HttpPort < PortMax)
+                {
+                    HttpPort++;
+                }
+
+                ShellProcess = Process.Start(PerfettoPluginConstants.TraceProcessorShellPath, $"-D --http-port {HttpPort} -i {tracePath}");
```

Comment:
Update so that this is launched from the correct directory and the shell.exe can be found